portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,27 +14,61 @@
 
 package com.google.sps.servlets;
 
+import com.google.gson.Gson;
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-import java.util.ArrayList;
-import com.google.gson.Gson;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  ArrayList<UserComment> userComments = new ArrayList<>();
+
+  /*
+   * Called when a client submits a GET request to the /data URL
+   * Displays all recorded user comments on page
+   */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    ArrayList<String> userComments = new ArrayList<>();
-    userComments.add("Nice website");
-    userComments.add("I like the teal color scheme");
-    userComments.add("Could be improved");
     Gson gson = new Gson();
     String json = gson.toJson(userComments);
     response.setContentType("application/json;");
     response.getWriter().println(json);
   }
+
+  /*
+   * Called when a client submits a POST request to the /data URL
+   * Adds submitted comment to internal record if the comment is
+   * non-empty. Then, the page is reloaded.
```

Comment:
You might want to think about how to avoid "page reloading" :) 

No need to do this in this PR (or this week), but it is a good exercise. I would say this use case is one of the main motivations for many frontend frameworks.

Reply:
I looked around a bit and it seems like the solution involves Ajax and JQuery. I don't know much about these so I'll look into this later in the week.

Reply:
Yeah you will need AJAX and "client-side render". JQuery is one way to achieve it.

As I said no need to do anything in this PR :)